pyteamcity/future/artifact.py

```diff
@@ -0,0 +1,93 @@
+import fnmatch
+import os
+
+from .core.utils import parse_date_string
+
+
+class Artifact(object):
+    def __init__(self, build, path=''):
+        self.build = build
+        self.path = path
+        teamcity = self.build.build_query_set.teamcity
+        url = self.build.api_url + '/artifacts/metadata/' + self.path
+        res = teamcity.session.get(url)
+        res.raise_for_status()
+        self._data = res.json()
+
+    @property
+    def name(self):
+        return self._data['name']
+
+    def getsize(self):
+        return self._data.get('size')
+
+    @property
+    def size(self):
+        return self.getsize()
+
+    @property
+    def modification_time(self):
+        return parse_date_string(self._data['modificationTime'])
+
+    def splitext(self):
+        return os.path.splitext(self.name)
+
+    @property
+    def ext(self):
+        return self.splitext()[1]
+
+    def fnmatch(self, pattern):
+        return fnmatch.fnmatch(self.name, pattern)
```

Comment:
My recommendation is to not eagerly implement methods that you expose as a public API. If you haven't put much thought into and and you make it public, well then you are stuck with it. Every time I implement code I ask myself 2 things:

1) do I have a use case for it?
2) if yes and it's a public API, is it really mature enough to be implemented knowing that if I need to change the signature later then I won't be able to make backwards incompatible changes easily.

You'd have to write tests for it too, which is additional work. But granted, this is not necessarily the riskiest API but still worth considering.